src/compas_timber/utils/__init__.py

```diff
@@ -290,4 +291,56 @@ def distance_segment_segment(segment_a, segment_b):
     return distance_point_point(pt_seg_a, pt_seg_b)
 
 
-__all__ = ["intersection_line_line_param", "intersection_line_plane_param", "intersection_line_beam_param", "classify_polyline_segments", "distance_segment_segment"]
+def is_polyline_clockwise(polyline, normal_vector):
+    """Check if a polyline is clockwise.
+
+    Parameters
+    ----------
+    polyline : :class:`compas.geometry.Polyline`
+        The polyline to check.
+
+    Returns
+    -------
+    bool
+        True if the polyline is clockwise, False otherwise.
+
+    """
+    assert polyline[0] == polyline[-1], "Polyline should be closed"
+    angle_sum = 0
+    for i in range(len(polyline) - 1):
+        u = Vector.from_start_end(polyline[i - 1], polyline[i])
+        v = Vector.from_start_end(polyline[i], polyline[i + 1])
+        angle = angle_vectors_signed(u, v, normal_vector)
+        angle_sum += angle
+    return angle_sum < 0
+
+
+def correct_polyline_direction(polyline, normal_vector, clockwise=False):
+    """Corrects the direction of a polyline to be counter-clockwise around a given vector.
+
+    Parameters
+    ----------
+    polyline : :class:`compas.geometry.Polyline`
+        The polyline to correct.
+
+    Returns
+    -------
+    :class:`compas.geometry.Polyline`
+        The corrected polyline.
+
+    """
+    cw = is_polyline_clockwise(polyline, normal_vector)
+    if (cw and not clockwise) or (not cw and clockwise):
```

Comment:
this is a logical XOR, you can simplify to:
```suggestion
    if cw ^ clockwise:
```

Reply:
interesting! thanks!